### PR TITLE
Issue/4480 reader add follow to post header

### LINKED ISSUE #4480: Reader: Add Follow button next to site info
This PR https://github.com/wordpress-mobile/WordPress-Android/pull/4475 moves the ellipsis to the bottom of the card. However, the Follow button is still hidden behind a click:

screenshot 2016-08-04 16 13 18

Should be:
screenshot-2016-08-04-16 14 58

The ellipsis can still contain a Follow button alongside a "Block this blog" option.

/cc @nbradbury

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #4480 - adds a Follow button to the header above posts in the reader list view. Note that the button isn't shown for "Followed Sites" or when there's a site header (ie: Discover or blog preview).

screenshot_1470422523

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java [hunk 1]
```diff
@@ -762,13 +787,44 @@ private void toggleLike(Context context, ReaderPostViewHolder holder, ReaderPost
...
+     */
+    private void toggleFollow(final Context context, final ReaderPost post) {
+        if (post == null || !NetworkUtils.checkConnection(context)) {
+            return;
+        }
+
+        boolean isCurrentlyFollowed = ReaderPostTable.isPostFollowed(post);
+        final boolean isAskingToFollow = !isCurrentlyFollowed;
+
+        ReaderActions.ActionListener actionListener = new ReaderActions.ActionListener() {
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java]: I noticed that `actionListener` is unused, and as a result error handling for `followBlogForPost` is not happening.

**Comment by khaykov** [comment]
Another really minor issue I noticed, is that you can repeatedly press follow button, and doing so might lead to discrepancies in "following" state between adapter and db, also `followBlogForPost` will sometimes produce errors (the toast will popup if you pass `actionListener` to it). Maybe adding some sort of progress state to the button, or just disabling clicks until response come from server will solve it.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java]: > I noticed that actionListener is unused, and as a result error handling for followBlogForPost is not happening.

Whoops, that was a silly one! Fixed in ffee003.

**Comment by nbradbury** [comment]
> Maybe adding some sort of progress state to the button, or just disabling clicks until response come from server will solve it.

Good call. Fixed in 0426cb5 and ready for another round.

**Comment by khaykov** [comment]
Thanks for the fixes, Nick! Works great! :shipit:

**Comment by jancavan** [comment]
Looks good as well 👍

**Pull request merged**